tests/consensus/devnet/fc/test_attestation_processing.py

```diff
@@ -0,0 +1,694 @@
+                            source_slot=Slot(0),  # Genesis
+                            target_slot=Slot(1),
+                            location="new",
+                        ),
+                    ],
+                ),
+            ),
+        ],
+    )
+
+
+def test_attestation_superseding_same_validator(
+    fork_choice_test: ForkChoiceTestFiller,
+) -> None:
+    """
+    Newer attestation from same validator supersedes older attestation.
+
+    Scenario
+    --------
+    Process blocks at slots 1 and 5 (same proposer: validator 1).
+
+    Expected:
+        - After slot 1: validator 1 attests to slot 1
+        - After slot 5: validator 1 attests to slot 5 (supersedes slot 1)
+
+    Why This Matters
+    ----------------
+    With round-robin proposer selection, slots 1 and 5 use the same validator.
+
+    When that validator proposes again, their newer attestation supersedes the older one.
+    Both dictionaries are keyed by validator index, so only the most recent
+    attestation per validator is retained.
+
+    Key insight: Attestations accumulate across validators but supersede within validators.
+    """
+    fork_choice_test(
+        steps=[
+            BlockStep(
+                block=BlockSpec(slot=Slot(1)),
+                checks=StoreChecks(
+                    head_slot=Slot(1),
+                    attestation_checks=[
+                        AttestationCheck(
+                            validator=ValidatorIndex(1),
+                            attestation_slot=Slot(1),
+                            head_slot=Slot(1),
+                            source_slot=Slot(0),
+                            target_slot=Slot(1),
+                            location="new",
+                        ),
+                    ],
+                ),
+            ),
+            BlockStep(
+                block=BlockSpec(slot=Slot(5)),
+                checks=StoreChecks(
+                    head_slot=Slot(5),
+                    attestation_checks=[
+                        # Validator 1's newer attestation (superseded the old one)
+                        AttestationCheck(
+                            validator=ValidatorIndex(1),
+                            attestation_slot=Slot(5),
+                            head_slot=Slot(5),
+                            target_slot=Slot(5),
+                            location="new",
+                        ),
+                    ],
+                ),
+            ),
+        ],
+    )
+
+
+def test_attestations_move_to_known_between_blocks(
+    fork_choice_test: ForkChoiceTestFiller,
+) -> None:
+    """
+    Attestations move from latest_new to latest_known between blocks.
+
+    Scenario
+    --------
+    Process blocks at slots 1 and 2 (different proposers: validators 1 and 2).
+
+    Expected:
+        - After slot 1: new attestations = 1, known attestations = 0
+        - After slot 2: new attestations = 1, known attestations = 1
```

Comment:
I think these 2 lines are a bit misleading because new/known attestations should take account of non-proposer attestations too and the test is not asserting the count